### ISSUE TITLE
feature request: firewall and gateway subnet

### ISSUE THREAD

**Issue by joakimhellum-in** [innovationnorway/terraform-azurerm-network#2]
It would be nice to be able to create the firewall and gateway subnet in the network module as well, for example:

```hcl
module "network" {
  source = "innovationnorway/network/azurerm"

  resource_group_name = "${azurerm_resource_group.main.name}"

  name = "westeurope-network"

  address_spaces = ["10.0.0.0/16"]

  # Specify the address prefix of the firewall subnet.
  firewall_subnet = "10.0.1.0/24"

  # Specify the address prefix of the gateway subnet.
  gateway_subnet = "10.0.2.0/24"

  private_subnets = ["10.0.10.0/24"]
}

```

- The firewall subnet must be named `AzureFirewallSubnet` to work properly, and minimum size is either /25 or /26.

- The gateway subnet must be named `GatewaySubnet` to work properly, and minimum size is /29 or /27.

References:
- [Do I need a 'GatewaySubnet'?](https://docs.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-vpn-faq#do-i-need-a-gatewaysubnet])

**Comment by antonbabenko** [comment]
Ok, I will add this feature. As far as I understand, names of subnets can be different if a user wants.

**Comment by joakimhellum-in** [comment]
Thanks alot. 

The names of the Firewall and Gateway subnet cannot be changed, and must be `AzureFirewallSubet` and `GatewaySubnet`. These names are hardcoded into Azure APIs, the portal and tools (such as Azure CLI). So there will be unexpected behavior and issues to associate resources unless the names are exactly `AzureFirewallSubnet` and `GatewaySubnet`. Sorry for this inconvenience.

![image](https://user-images.githubusercontent.com/34519450/51425920-33acce80-1be3-11e9-8c4d-af858dd84603.png)

**Comment by joakimhellum-in** [comment]
Yes, the examples and documentation for the Firewall is not very clear about this.

Reference to https://docs.microsoft.com/en-us/azure/firewall/tutorial-firewall-deploy-portal
It is not very explicit in the documentation at all that the name of the subnet needs to be `AzureFirewallSubnet`. It is only mentioned in another context here:
![image](https://user-images.githubusercontent.com/34519450/51426294-dc116180-1be8-11e9-9aa2-07e4ea827c85.png)

Here is from the API specifications that will validate the name of the subnet:
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/azureFirewall.json#L268

It is also documented in some of the SDKs as well. But for the Gateway subnet, this is in the FAQ of the official docs.

**Comment by antonbabenko** [comment]
Added support for Azure Firewall in this module in [v0.4.0](https://github.com/innovationnorway/terraform-azurerm-network/releases/v0.4.0), which was just released.

**Comment by joakimhellum-in** [comment]
Very nice, thank you :)

**Comment by antonbabenko** [comment]
Reopening, because gateway subnet is not there yet. I am working on it now.

**Comment by antonbabenko** [comment]
This is now released in [v0.5.0](https://github.com/innovationnorway/terraform-azurerm-network/releases/v0.5.0).
